### PR TITLE
[intel-npu] Adding EXECUTION_DEVICES read-only property

### PR DESCRIPTION
**Pull request by csoka**
### Details:
 - Adding EXECUTION_DEVICES read-only property for intel-npu plugin.

### Tickets:
 - EISW-112990

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/src/plugin.cpp [hunk 1]
```diff
@@ -352,6 +352,12 @@ Plugin::Plugin()
           [&](const Config& config) {
               return _metrics->GetGops(get_specified_device_name(config));
           }}},
+        {ov::execution_devices.name(),
+         {true,
+          ov::PropertyMutability::RO,
+          [&](const Config& config) {
+              return std::string("NPU");
```
Comment (ilya-lavrenov) [inline src/plugins/intel_npu/src/plugin/src/plugin.cpp:359]: if you have multiple device, it should be concrete device with so-called `device_id`

Comment (csoka) [reply src/plugins/intel_npu/src/plugin/src/plugin.cpp]: added device_id to the string. thanks!

File: src/plugins/intel_npu/src/plugin/src/plugin.cpp [hunk 2]
```diff
@@ -352,6 +352,12 @@ Plugin::Plugin()
           [&](const Config& config) {
               return _metrics->GetGops(get_specified_device_name(config));
           }}},
+        {ov::execution_devices.name(),
+         {true,
+          ov::PropertyMutability::RO,
+          [&](const Config& config) {
+              return std::string("NPU" + config.get<DEVICE_ID>());
```
Comment (ilya-lavrenov) [inline src/plugins/intel_npu/src/plugin/src/plugin.cpp:359]: device id is added via "." symbols, e.g. `NPU.1`

**Review by sbutnari — APPROVED** [review]

File: src/plugins/intel_npu/src/plugin/src/plugin.cpp (hunk 2 above)
Comment (csoka) [reply src/plugins/intel_npu/src/plugin/src/plugin.cpp]: updated now, thanks!

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**